File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Go/s452855095_commented.go
package main

import (
	"fmt"
)

func main() {
	var n int
	fmt.Scan(&n)
	arr := make([]int, n)
	for i := range arr {
		fmt.Scan(&arr[i])
	}
	cnt := 0
	for {
		flag := true
		for i := 0; i < n; i++ {
			if arr[i]%2 != 0 {
				flag = false
				break
			}
		}
		if !flag {
			break
		}
		for i := 0; i < n; i++ {
			arr[i] /= 2
		}
		cnt++
	}
	fmt.Println(cnt)
	// End of Code
}

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/C/s452855095_commented.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int n;
    scanf("%d", &n);
    int *arr = (int*)malloc(n * sizeof(int));
    for(int i = 0; i < n; i++) {
        scanf("%d", &arr[i]);
    }
    int cnt = 0;
    while (1) {
        int flag = 1; // Reset flag each iteration
        for(int i = 0; i < n; i++) {
            if (arr[i] % 2 != 0) {
                flag = 0;
                break;
            }
            arr[i] /= 2;
        }
        if (flag) {
            cnt++;
        } else {
            break;
        }
    }
    printf("%d\n", cnt);
    free(arr);
    return 0;
}
// End of Code
